subprojects/model-core/src/test/groovy/org/gradle/internal/properties/annotations/NestedBeanAnnotationHandlerTest.groovy

```diff
@@ -66,51 +63,13 @@ class NestedBeanAnnotationHandlerTest extends Specification {
 
     def "absent optional nested property is ignored"() {
         when:
-        handler.visitPropertyValue("name", value, propertyMetadata, propertyVisitor, context)
+        handler.visitPropertyValue("name", value, propertyMetadata, propertyVisitor)
 
         then:
         1 * value.call() >> null
         1 * propertyMetadata.isAnnotationPresent(TestOptional) >> true
         0 * _
     }
-
-    def "exception thrown by nested property is propagated"() {
-        PropertyValue validatingValue = null
-        def exception = new RuntimeException("BOOM!")
-
-        when:
-        handler.visitPropertyValue("name", value, propertyMetadata, propertyVisitor, context)
-
-        then:
-        1 * value.call() >> {
-            throw exception
-        }
-        1 * propertyVisitor.visitInputProperty("name", _, false) >> { arguments ->
-            validatingValue = arguments[1]
-        }
-        0 * _
-
-        when:
-        validatingValue.call()
-
-        then:
-        0 * _
-        def thrown = thrown(RuntimeException)
-        exception == thrown
-    }
-
-    def "nested bean is added"() {
```

Comment:
💭 We don't do this in the annotation handler anymore, but any test checking `@Nested` functionality would fail if this didn't work.

Reply:
🤔 Would it make sense to add a test for `AbstractNestedRuntimeBeanNode` now to check the `@Nested` handling?

Reply:
I don't see much value in doing so. We are covered by integration tests from the functionality perspective. We are about to revamp the whole property discovery process that will remove/replace most of those unit tests anyway.